src/libraries/System.Net.Sockets/tests/FunctionalTests/ReceiveMessageFrom.cs

```diff
@@ -256,6 +256,26 @@ public async Task ShutdownSend_ReceiveFromShouldSucceed()
             var r = await ReceiveMessageFromAsync(receiver, new byte[1], sender.LocalEndPoint);
             Assert.Equal(1, r.ReceivedBytes);
         }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)] // MSG_BCAST is Windows-specifc
```

Comment:
They should, and based on a quick look at the code, they do.

This test however works with `SocketFlags.Broadcast` (`MSG_BCAST`) which is Windows-specific. I wasn't able to construct a cross platform test that works with other flags. For that I would need for example enforce the truncation of a UDP packet (`MSG_TRUNC`). Maybe @tmds has an idea, but this is not worth wasting too much time on IMO.

Reply:
> I wasn't able to construct a cross platform test that works with other flags. For that I would need for example enforce the truncation of a UDP packet (MSG_TRUNC). Maybe @tmds has an idea, but this is not worth wasting too much time on IMO

To get `MSG_TRUNC` you can make the receive buffer smaller than the message that was sent.

Reply:
> To get `MSG_TRUNC` you can make the receive buffer smaller than the message that was sent.

If you mean using smaller receive buffer than the original packet payload sent, yeah I tried, but that did not result in `MSG_TRUNC` on Ubuntu (I guess it means that the rest of the packet is in the OS buffer so there was no truncation actually). Also tried reducing `SO_RECVBUF` but it had no effect (Maybe it does not affect buffer sizes in case of UDP?)

> maybe you can add Linux specific test as complement....?

Cross-platform & Linux-specific would be both great, but I wasn't able to come up with anything.

Reply:
> that did not result in MSG_TRUNC on Ubuntu

What does this give on Ubuntu?

```cs
using System.Net;
using System.Net.Sockets;
using System.Text;

using var receiver = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp);
receiver.Bind(new IPEndPoint(IPAddress.Any, 0));

using var sender = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp);

sender.SendTo(Encoding.UTF8.GetBytes("Hello world!"), receiver.LocalEndPoint!);

SocketFlags socketFlags = SocketFlags.None;
EndPoint endPoint = sender.LocalEndPoint!;
receiver.ReceiveMessageFrom(new byte[1], ref socketFlags, ref endPoint, out _);
Console.WriteLine(socketFlags);
```

On Fedora, it gives me:
```
$ dotnet run
Truncated
```

Reply:
@antonfirsov you should be able to get `MSG_TRUNC` as described above.